src/lib/async-replace.ts

```diff
@@ -55,6 +53,7 @@ export const asyncReplace =
             // previous value display until we can replace it.
             if (i === 0) {
               part.clear();
+              itemPart.appendIntoPart(part);
```

Comment:
Why wait to append this?

Reply:
It's no longer sharing the start and end node with the parent part, so if we clear the parent after appending, we'll nuke this part. I think this is easier to understand than the previous sharing of markers. It's a little more intuitive that if a part contains a child part and you clear the parent, you'll remove the child.

Reply:
Are there more nodes now?

Reply:
And is it worth documenting (at least internally) how the marker nodes are setup between parts?

Reply:
This is one reason for always using the attachment methods rather than having calling code manipulate the markers directly. Between insert/append/clear, most code should not have to care about them.

Reply:
and I do think the "contains" semantic between parent and child parts is way more understandable here. It would have been weird if you cleared a parent and a single child part wasn't removed, or if you had multiple parts they were just broken.